frame/vesting/src/lib.rs

```diff
@@ -61,41 +60,120 @@ use frame_support::traits::{
 use frame_system::{ensure_signed, ensure_root, pallet_prelude::*};
 pub use weights::WeightInfo;
 pub use pallet::*;
+pub use vesting_info::*;
 
 type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
 type MaxLocksOf<T> = <<T as Config>::Currency as LockableCurrency<<T as frame_system::Config>::AccountId>>::MaxLocks;
 
 const VESTING_ID: LockIdentifier = *b"vesting ";
 
-/// Struct to encode the vesting schedule of an individual account.
-#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug)]
-pub struct VestingInfo<Balance, BlockNumber> {
-	/// Locked amount at genesis.
-	pub locked: Balance,
-	/// Amount that gets unlocked every block after `starting_block`.
-	pub per_block: Balance,
-	/// Starting block for unlocking(vesting).
-	pub starting_block: BlockNumber,
-}
+// Module to enforce private fields on `VestingInfo`
+mod vesting_info {
+	use super::*;
+	/// Struct to encode the vesting schedule of an individual account.
+	#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug)]
+	pub struct VestingInfo<Balance, BlockNumber> {
+		/// Locked amount at genesis.
+		locked: Balance,
+		/// Amount that gets unlocked every block after `starting_block`.
+		per_block: Balance,
+		/// Starting block for unlocking(vesting).
+		starting_block: BlockNumber,
+	}
 
-impl<
-	Balance: AtLeast32BitUnsigned + Copy,
-	BlockNumber: AtLeast32BitUnsigned + Copy,
-> VestingInfo<Balance, BlockNumber> {
-	/// Amount locked at block `n`.
-	pub fn locked_at<
-		BlockNumberToBalance: Convert<BlockNumber, Balance>
-	>(&self, n: BlockNumber) -> Balance {
-		// Number of blocks that count toward vesting
-		// Saturating to 0 when n < starting_block
-		let vested_block_count = n.saturating_sub(self.starting_block);
-		let vested_block_count = BlockNumberToBalance::convert(vested_block_count);
-		// Return amount that is still locked in vesting
-		let maybe_balance = vested_block_count.checked_mul(&self.per_block);
-		if let Some(balance) = maybe_balance {
-			self.locked.saturating_sub(balance)
-		} else {
-			Zero::zero()
+	impl<Balance: AtLeast32BitUnsigned + Copy, BlockNumber: AtLeast32BitUnsigned + Copy>
+		VestingInfo<Balance, BlockNumber>
```

Comment:
Should we expose merge vesting schedules as a method of `VestingInfo`? Downside is that it needs to be used with care because it assumes schedules have been vested up through the current block prior to merging.

Reply:
We might as well leave it up to the existing `merge3_schedules` function.